src/libraries/System.Numerics.Tensors/tests/TensorPrimitivesTests.cs

```diff
@@ -1431,7 +1454,27 @@ public static void MaxMagnitude_Tensor(int tensorLength)
                 maxMagnitude = MathFMaxMagnitude(maxMagnitude, i);
             }
 
-            AssertEqual(maxMagnitude, TensorPrimitives.MaxMagnitude(x));
+            Assert.Equal(maxMagnitude, TensorPrimitives.MaxMagnitude(x));
+            Assert.Equal(SingleToUInt32(x[TensorPrimitives.IndexOfMaxMagnitude(x)]), SingleToUInt32(TensorPrimitives.MaxMagnitude(x)));
+        }
+
+        [Theory]
+        [MemberData(nameof(TensorLengths))]
+        public static void MaxMagnitude_Tensor_SpecialValues(int tensorLength)
+        {
+            using BoundedMemory<float> x = CreateAndFillTensor(tensorLength);
+
+            RunForEachSpecialValue(() =>
+            {
+                float maxMagnitude = x[0];
+                foreach (float i in x.Span)
```

Comment:
nit:
```suggestion
                foreach (float f in x.Span)
```

I see we have this existing convention in other magnitude tests, but it didn't make sense to me.  Why do we use `i` to represent a float?